muffnn/mlp/base.py

```diff
@@ -247,14 +250,39 @@ def _set_up_graph(self):
                 tf.placeholder(np.float32, [None], "input_values")
             self._input_shape = \
                 tf.placeholder(np.int64, [2], "input_shape")
+
+            # TODO: understand how to construct a pipeline for sparse data
+            sparse_dataset = tf.data.Dataset.from_tensor_slices((self._input_indices, self._input_values, self._input_shape ))
+            # TODO: it may be possible to use map
+            sparse_dataset = sparse_dataset.map(lambda x: tf.SparseTensor(*x))
+            labels_dataset = tf.data.Dataset.from_tensor_slices((self._labels_placeholder))
+
+            dataset = tf.data.Dataset.zip((sparse_dataset, labels_dataset))
+            dataset = dataset.shuffle(10000).batch(self.batch_size).repeat(self.n_epochs)
+
+            iterator = dataset.make_initializable_iterator()
+
+            # TODO: pray the following is not what we have to do
+            (x, y, z), self._input_targets = iterator.get_next()
+
             # t will be the current layer as we build up the graph below.
-            t = tf.SparseTensor(self._input_indices, self._input_values,
-                                self._input_shape)
+            t = tf.SparseTensor(x, y, x)
+            # TODO: if map works
+            self._input_values, self._input_targets = iterator.get_next()
```

Comment:
Per our discussion I think we just use the simple version of this for now and can change it later if needed.

Reply:
The only issue is that we need to keep the dataset function chaining inside the `is_sparse` block so that we can set `t` correctly

Reply:
right yep.